ckanext/geocat/commands.py

```diff
@@ -1,101 +1,112 @@
 import sys
 from pprint import pprint
-import ckan.lib.cli
-import ckanext.geocat.metadata as md
-import ckanext.geocat.xml_loader as loader
+from ckan.lib.cli import CkanCommand
+from ckanext.geocat.utils import csw_processor, csw_mapping
+from ckanext.geocat.harvester import DEFAULT_PERMA_LINK_URL, DEFAULT_PERMA_LINK_LABEL  # noqa
 
 
-class GeocatCommand(ckan.lib.cli.CkanCommand):
+class GeocatCommand(CkanCommand):
 
     '''Command to query geocat
 
     Usage::
-
-            paster geocat search birds
-            paster geocat cql "csw:AnyText like '%birds%'"
-            paster geocat list "keyword = 'opendata.swiss'" https://www.geocat.ch/geonetwork/srv/eng/csw-ZH/
-            paster geocat dataset "8ae7eeb1-04d4-4c78-93e1-4225412db6a4" https://www.geocat.ch/geonetwork/srv/eng/csw-ZH/
-
+    
+    With this command you can query a remote csw source:
+    
+    'paster geocat list https://www.geocat.ch/geonetwork/srv/eng/csw-ZH/'
+    
+    The 'list'  command will bring back the record ids of the remote source.
+    You can also add query arguments to it: this here for example will search
+    for the keyword 'opendata.swiss' on the remote source:
+    
+    'paster geocat list https://www.geocat.ch/geonetwork/srv/eng/csw-ZH/ --key=keyword --term=opendata.swiss'
+    
+    Once you have the record ids, you can map a specific remote record: 
+    
+    'paster geocat dataset https://www.geocat.ch/geonetwork/srv/eng/csw-ZH/ 8ae7eeb1-04d4-4c78-93e1-4225412db6a4'
+    
+    The 'dataset' command uses the same mapping as the harvester, except some extras as the geocat-permalink and 
+    some other fields that are taken from the harvester config when harvesting and just get defaults when
+    the command is used. 
     '''  # noqa
     summary = __doc__.split('\n')[0]
     usage = __doc__
-    DEFAULT_CSW_SERVER = 'http://www.geocat.ch/geonetwork/srv/eng/csw'
-    DEFAULT_CQL = "keyword = 'opendata.swiss'"
+
+    def __init__(self, name):
+        super(CkanCommand, self).__init__(name)
+        self.parser.add_option(
+            '--query', action="store", type="string",  dest='cql_query',
+            default=csw_processor.CQL_QUERY_DEFAUL,
+            help='key for cql search')
+        self.parser.add_option(
+            '--term', action="store", type="string",  dest='cql_term',
+            default=csw_processor.CQL_SEARCH_TERM_DEFAUT,
+            help='searchterm for cql search')
 
     def command(self):
+        self._load_config()
         options = {
-            'search': self.searchCmd,
-            'cql': self.cqlCmd,
```

Comment:
Because I changed the way that the search is performed: it used to be a complete cql query, Now I am using `owslib.fes.PropertyIsEqualTo`. It takes to arguments: the property `keyword` and the argument `opendata.swiss`. It felt safer to me that way and it was what found on the owslib tutorial. 

I did not want to generalize the the csw processor just to serve these commands that are almost never used by us. I can understand that initially it felt good to add very general commands to the harvester for convenience, but now when I work with Geocat data myself, I have mostly 2 use cases: check the data that the harvester gets or get on special record. 

I adapted the `list` command so that it also takes a query argument such as `keyword` and a searchterm such as `opendata.swiss`, but I dropped the more general `cql` command since the csw processor would not serve that command any more in its current form.

Does that sound reasonable enough to you.

Reply:
OK by me :)